### ISSUE TITLE
Maintain query parameter ordering in QueryParams

### ISSUE THREAD

**Issue by jdegoes** [zio/zio-http#2586]
```scala
trait QueryParams {
  def keys: Iterable[String]

  def get(key: String): Option[Chunk[String]]

  def getOrElse(key: String, default: Chunk[String]): Chunk[String]
}
```

With a `ListMap` implementation and use of it on ingest from Netty to preserve ordering.

**Comment by easel** [comment]
/attempt #2586

<details id="algora-options">
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clr2hj5380004l40fbcw8tnoj/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by easel** [comment]
This one should be fixed with the merge of #2591
